gratipay/models/participant/email.py

```diff
@@ -233,31 +233,36 @@ def _set_primary_email(self, email, cursor):
 
 
     def finish_email_verification(self, email, nonce):
-        """Given an email address and a nonce as strings, return a tuple:
-        (a ``VERIFICATION_*`` constant, and a list of packages for
-        ``VERIFICATION_SUCCEEDED`` or ``None`` otherwise)
+        """Given an email address and a nonce as strings, return a tuple: (a
+        ``VERIFICATION_*`` constant, a list of packages for
+        ``VERIFICATION_SUCCEEDED`` or ``None`` otherwise, and a boolean
+        indicating whether the participant's PayPal address was updated)
         """
         if '' in (email.strip(), nonce.strip()):
-            return VERIFICATION_FAILED, None
+            return VERIFICATION_FAILED, None, None
         with self.db.get_cursor() as cursor:
             record = self.get_email(email, cursor, and_lock=True)
             if record is None:
-                return VERIFICATION_FAILED, None
+                return VERIFICATION_FAILED, None, None
             packages = self.get_packages_claiming(cursor, nonce)
             if record.verified and not packages:
                 assert record.nonce is None  # and therefore, order of conditions matters
-                return VERIFICATION_REDUNDANT, None
+                return VERIFICATION_REDUNDANT, None, None
             if not constant_time_compare(record.nonce, nonce):
-                return VERIFICATION_FAILED, None
+                return VERIFICATION_FAILED, None, None
             if (utcnow() - record.verification_start) > EMAIL_HASH_TIMEOUT:
-                return VERIFICATION_FAILED, None
+                return VERIFICATION_FAILED, None, None
             try:
+                paypal_updated = False
                 if packages:
                     self.finish_package_claims(cursor, nonce, *packages)
                 self.save_email_address(cursor, email)
+                if packages and not self.get_payout_routes(good_only=True):
```

Comment:
Done in 91a567cdcd69a614b9d26784aba665f5aac63c57. `has_payout_route` includes failing routes, and we only want non-failing ones here.